ecs-init/docker/dependencies.go

```diff
@@ -65,13 +64,98 @@ func (client godockerClientFactory) NewVersionedClient(endpoint string, apiVersi
 	return godocker.NewVersionedClient(endpoint, apiVersionString)
 }
 
+// backupAPIVersions are API versions that have been deprecated on the most recent
+// versions of the docker engine, and are only checked if none of the preferredAPIVersions
+// are available.
+func backupAPIVersions() []string {
+	return []string{
+		"1.21",
+		"1.22",
+		"1.23",
+		"1.24",
+	}
+}
```

Comment:
IIUC before this change ecs-init was not compatible with environments with Docker engines that do not support API versions >= 1.25. But this PR makes ecs-init compatible with such environments. So, does that mean the older API versions were artificially made incompatible with ecs-init?

I understand that Task ENI is not supported on API versions < 1.25, but ecs-init doesn't actually make any Docker API calls that are not supported on API versions < 1.25?

Reply:
> ecs-init doesn't actually make any Docker API calls that are not supported on API versions < 1.25?

Not by default, but I will check on the various flags that ecs-init can add during startup to see if there are any that are not supported before docker 1.25.